src/Auth.ts

```diff
@@ -0,0 +1,127 @@
+import * as jwt from 'jsonwebtoken'
+
+// Types
+import { IAuth, IToken } from '../types'
+
+// Constants
+import { API_URL, LOGIN_API, LOGOUT_API, REFRESH_API } from './constants'
+
+// Services
+import * as authService from './services/auth'
+
+export default class Auth {
+  public accessToken: string = ''
+  public ethAddress: string = ''
+  public password: string = ''
+  public refreshToken: string = ''
+
+  constructor(ethAddress?: string, password?: string) {
+    if (!ethAddress && !password) {
+      this.getCredentials()
+    } else {
+      this.ethAddress = ethAddress as string
+      this.password = password as string
+    }
+  }
+
+  public getCredentials() {
+    const data = authService.getCredentials() as IAuth
+
+    if (Object.keys(data).length) {
```

Comment:
Instead of checking the length of the object can we check if `data` contains a meaningful property that can be checked against to see if the obj has required properties?

Reply:
It's unnecessary validation when you're trying to read from the temp file whose contents have been saved by the CLI